Docs cleanup

dvdherron opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&swag)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->



## Description
A lot of changes were made for the upcoming release. This PR tries to make sure that
- All changes and additions have been documented and accounted for in the changelog(s)
- Links are up to date and point to the correct places



## Steps to test/reproduce

I've been scanning related [closed PRs](https://github.com/oddbird/accoutrement/pulls?q=is%3Apr+is%3Aclosed) to get an overview of what has changed. 

These are the ones where the bulk of the changes were made:

1. [Modules Core](https://github.com/oddbird/accoutrement/pull/76)
2. [Modules Init](https://github.com/oddbird/accoutrement/pull/89)
3. [Type](https://github.com/oddbird/accoutrement/pull/90)
4. [Color](https://github.com/oddbird/accoutrement/pull/91)
5. [Scale](https://github.com/oddbird/accoutrement/pull/92)
6. [Layout](https://github.com/oddbird/accoutrement/pull/93)
7. [Animate](https://github.com/oddbird/accoutrement/pull/95)


_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

dvdherron commented:
![](https://github.trello.services/images/mini-trello-icon.png) [I want Accoutrement on Sass Modules](https://trello.com/c/9tyINy2o/23-i-want-accoutrement-on-sass-modules)

dvdherron commented on sass/tokens/register/_index.scss:
```diff
@@ -1,4 +1,4 @@
-/// # Registered Functions
+/// # Pre-Registered Functions
```
@mirisuzanne Should the 'math' functions in `tokens/register` show in the docs? I think this is the only module with a nested folder. Is that what's causing 'Pre-Registered Functions' to not show in the docs?

dvdherron commented:
@mirisuzanne @stacyk This is ready to start reviewing. 

Some questions and known issues:
- Do we want to remove all Eyeglass references? I think all mentions of it in the modules are removed, there are just references in the README and the Changelog. 
- Should the "Pre-Registered Functions" section show in the public docs? I left a question above about why that might not be working. 
- In the `map-compile-with` example (under `tokens/compile`) I'm not sure why the compiled result is returning `null`

If you you notice anything else missing or or weird just let me know

mirisuzanne commented:
@dvdherron 

>     * Do we want to remove all Eyeglass references? I think all mentions of it in the modules are removed, there are just references in the README and the Changelog.

I don't know. We still support eyeglass in the npm package. It just seemed weird to highlight it, as one of many possible package managers/bundlers? Maybe there's some way to just note in one place that tools like eyeglass or webpack will impact the import path? @jgerigmeyer may have better ideas how/where to note that?
 
>     * Should the "Pre-Registered Functions" section show in the public docs? I left a question above about why that might not be working.

I… can't figure this out. Nested folders shouldn't make any difference here, but yes - when I move the files, they suddenly appear in the docs. @jgerigmeyer any idea why that might be? We can move all the files, I'd just like to understand what's going on.
 
>     * In the `map-compile-with` example (under `tokens/compile`) I'm not sure why the compiled result is returning `null`

- The example runs a map through the `color()` function.
- The color function looks for each of those color names in the `tools.$colors` map
- The colors are not defined in `tools.$colors`, only in the example map

There are two possible solutions: 

- change `$brand-colors` to be `tools.$colors`, so that the color function can find them
- pass in `$brand-colors` as a `$source` argument, the same way we do in `compile-colors`

Probably the first option is simplest?

> If you you notice anything else missing or or weird just let me know

I'll do a full review here in a bit…

jgerigmeyer commented:
> > Do we want to remove all Eyeglass references? I think all mentions of it in the modules are removed, there are just references in the README and the Changelog.
> 
> I don't know. We still support eyeglass in the npm package. It just seemed weird to highlight it, as one of many possible package managers/bundlers? Maybe there's some way to just note in one place that tools like eyeglass or webpack will impact the import path? @jgerigmeyer may have better ideas how/where to note that?

I don't have a strong feeling about this. It seems like what's there isn't hurting anything (although the Eyeglass example in the README seems identical to the non-Eyeglass example), but I'm also fine removing the specifics since Eyeglass/Webpack users would likely know how to import npm packages. 🤷 

> > Should the "Pre-Registered Functions" section show in the public docs? I left a question above about why that might not be working.
> 
> I… can't figure this out. Nested folders shouldn't make any difference here, but yes - when I move the files, they suddenly appear in the docs. @jgerigmeyer any idea why that might be? We can move all the files, I'd just like to understand what's going on.

Fixed in https://github.com/oddbird/accoutrement/pull/97/commits/2f0e2d16bb24f52fcab9334a9264df8888103472 (see https://medium.com/@jakubsynowiec/you-should-always-quote-your-globs-in-npm-scripts-621887a2a784).

stacyk commented on CHANGELOG.md:
```diff
@@ -2,39 +2,40 @@
 
   - NEW: `compile-*` functions provide an export option for respective token maps
-    (see: `compile-changes`, `compile-easing`, `compile-times`).
+    (see: `compile-changes()`, `compile-easing()`, `compile-times()`).
```
Maybe we can add links to these since we say "see this thing"? It will make it a little easier for someone to find them. I realize we aren't linking to all new or revised functions, but we probably could?

dvdherron replied:
@stacyk Ah, yes. Adding links definitely make sense here and I think it makes to the changlelog more user friendly in general to add them to other references someone might want to jump to. I added in these links for now and will add more for the recent changes.

stacyk replied:
Looks good and was easier to use 🎉

stacyk commented on CHANGELOG.md:
```diff
@@ -49,15 +50,26 @@ and removing some features that no longer seem necessary.
   browser normalization, is no longer included. We now recommend
   using [CSS Remedy](https://github.com/jensimmons/cssremedy).
```
Do we want to address the spacing issues that we are seeing on projects that are using this init-free version of Accoutrement?

dvdherron replied:
I think it could be worth it to add a note mentioning the specific layout issues we've noticed, and/or a general note saying "this might break layouts" . . . I'll try to gather what we've noticed so far and add that in.

stacyk commented on CHANGELOG.md:
```diff
@@ -49,15 +50,26 @@ and removing some features that no longer seem necessary.
-  "tokens" (the map syntax parser),
-  "variables" (converting Sass variables & maps to CSS custom properties),
-  and "ratios" (for typographic scales and layout aspect ratios)
+  for [utilities][utils] (extending several Sass modules),
+  [tokens][token] (the map syntax parser),
+  [variables][vars] (converting Sass variables & maps to CSS custom properties),
+  and [ratios][ratio] (for typographic scales and layout aspect ratios)
```
All of these links are currently broken, though I think they will work once this is fully published since they are the new pages.

dvdherron replied:
yes, these should work once this is fully published. That will be something good to test right after we merge.

stacyk reviewed (approved):
@dvdherron Thanks for compiling all this info together. I left a few optional comments so I am going to mark this as approved on my end even though I know there is still some work that may need to be done based on the conversations taking place.

dvdherron commented:
> @dvdherron
> 
> > Do we want to remove all Eyeglass references? I think all mentions of it in the modules are removed, there are just references in the README and the Changelog.
> 
> I don't know. We still support eyeglass in the npm package. It just seemed weird to highlight it, as one of many possible package managers/bundlers? Maybe there's some way to just note in one place that tools like eyeglass or webpack will impact the import path? @jgerigmeyer may have better ideas how/where to note that?
> 

Yeah, like Jonny said I guess having at least one mention of it doesn't hurt. Since we removed on all of the modules I just wanted to double check that we hadn't merely overlooked the remaining references. 

> > Should the "Pre-Registered Functions" section show in the public docs? I left a question above about why that might not be working.
> 
> I… can't figure this out. Nested folders shouldn't make any difference here, but yes - when I move the files, they suddenly appear in the docs. @jgerigmeyer any idea why that might be? We can move all the files, I'd just like to understand what's going on.
> 

Jonny's fix got these to show. I'll double check to make sure everything there is looking okay. 

> > In the `map-compile-with` example (under `tokens/compile`) I'm not sure why the compiled result is returning `null`
> 
>     * The example runs a map through the `color()` function.
> 
>     * The color function looks for each of those color names in the `tools.$colors` map
> 
>     * The colors are not defined in `tools.$colors`, only in the example map
> 
> 
> There are two possible solutions:
> 
>     * change `$brand-colors` to be `tools.$colors`, so that the color function can find them
> 
>     * pass in `$brand-colors` as a `$source` argument, the same way we do in `compile-colors`
> 
> 
> Probably the first option is simplest?
> 

Makes sense. Cleaned this up and it's compiling correctly now. 

> > If you you notice anything else missing or or weird just let me know
> 
> I'll do a full review here in a bit…

👍🏽

mirisuzanne commented on CHANGELOG.md:
```diff
@@ -47,64 +56,100 @@ and removing some features that no longer seem necessary.
 
   - NEW: These functions are now available directly,
     as well as being registered in the map syntax
+  - BREAKING: Private function `_a_list-template()` renamed `template()`
```
For functions that are still private, we can leave them out of this documentation entirely. For functions that used to be private, but are now public, we can consider them `new` rather than `breaking`.

dvdherron replied:
Removed this and other private functions. Changed the functions that are newly public to `NEW` and added descriptions.

mirisuzanne commented on CHANGELOG.md:
```diff
@@ -47,64 +56,100 @@ and removing some features that no longer seem necessary.
+  - NEW: [`has-token()`](token-inspect.html#function--has-token) function checks a map
     to see if the given token is defined,
     and supports the `outer->inner` nested token syntax.
+  - BREAKING: `_a_plus()` renamed [`plus()`](token-internal.html#function--plus)
+  - BREAKING: `_a_minus()` renamed [`minus()`](token-internal.html#function--minus)
+  - BREAKING: `_a_times()` renamed [`times()`](token-internal.html#function--times)
+  - BREAKING: `_a_divide()` renamed [`divide()`](token-internal.html#function--divide)
+  - BREAKING: `_a_modulo()` renamed [`modulo()`](token-internal.html#function--modulo)
```
These can also be documented as new functions, since they were previously private.

dvdherron replied:
Changed these to `NEW` functions

mirisuzanne commented on sass/color/_config.scss:
```diff
@@ -24,6 +26,17 @@ $contrast-light: white !default;
 
+// Contrast Settings
+// --------------
+///
```
```suggestion
/// Access the configurable contrast variables as an Accoutrement token map.
/// @access private
```

I don't think we need to expose this as part of the public API, it's really meant for internal use.

dvdherron replied:
made this private

mirisuzanne commented on sass/tokens/_index.scss:
```diff
@@ -65,18 +65,11 @@
 
 /// ## Import
-/// If you already imported `accoutrement/sass/tools` you are ready to go.
+/// If you've already imported `accoutrement/sass/tools` you are ready to go.
```
(see earlier comment about requiring `/sass/tools`)

dvdherron replied:
removed all of these

mirisuzanne commented on sass/tokens/_parse.scss:
```diff
@@ -256,7 +256,7 @@ $_stack: ();
       $lookup: lookup-alias($map, $alias-key);
 
-      // Replace #alias when lookup succeeds. Otherwise leave it in the string (eg #aaa)
+      // Replace #alias when lookup succeeds. Otherwise leave it in the string (e.g.#aaa)
```
```suggestion
      // Replace `#alias` when lookup succeeds. Otherwise leave it in the string (e.g. `#aaa`)
```

mostly just the missing space, but also added code fencing?

mirisuzanne commented on sass/tokens/register/_math.scss:
```diff
@@ -4,12 +4,15 @@
 /// `'plus'`, `'add'`, or `'+'`.
 ///
+/// @since 4.0.0 -
+/// - BREAKING: Renamed from `_a_plus` to `plus`
```
When we move something from private to public, we don't have to document it as a breaking change - just a new feature.

mirisuzanne commented on sass/type/_config.scss:
```diff
@@ -70,7 +70,7 @@ $font-path: '' !default;
 /// - NEW: Supports `'bold italic'` weight/style syntax
 ///   in addition to `('bold', 'italic')`
-/// - BREAKING: Full support for the [core](core.html) map-reference syntax,
+/// - BREAKING: Full support for the [core](tokens.html) map-reference syntax,
```
```suggestion
/// - BREAKING: Full support for the [token](tokens.html) map-reference syntax,
```

dvdherron replied:
fixed

mirisuzanne commented on sass/type/_fonts.scss:
```diff
@@ -22,6 +22,8 @@
 /// Retrieve and normalize font data from the `$fonts` map
 ///
+/// @since 4.0.0 -
+/// - BREAKING: Renamed `font`
```
Might be good to include the previous name here, so it's easy to search/replace

dvdherron replied:
good catch. this was a mistake. This function was not renamed. Removed the note.

mirisuzanne commented on sass/type/_index.scss:
```diff
@@ -15,7 +15,7 @@
 
 /// ## Import
-/// If you already imported `accoutrement/sass/tools` you are ready to go.
+/// If you've already imported `accoutrement/sass/tools` you are ready to go.
```
(see similar)

mirisuzanne reviewed (commented):
Wow, this is excellent. I left a few comments on small cleanup, but overall very thorough.

dvdherron commented on CHANGELOG.md:
```diff
@@ -2,39 +2,40 @@
-  - NEW: `compile-fonts` function provides an export option for font token maps.
+  - NEW: `compile-fonts()` function provides an export option for font token maps.
 
 ### 4.0.0-beta.2 - 12/03/21
```
Merged the two release notes together. We just need to add a date whenever we release.

mirisuzanne reviewed (approved):
looks great!

jgerigmeyer commented on CHANGELOG.md:
```diff
@@ -2,110 +2,144 @@
+  Legacy versions of Sass (Node & Ruby) are no longer supported.
+- BREAKING: The old "init" module, which provided lightweight
+  browser normalization, is no longer included. We now recommend
+  using [CSS Remedy](https://github.com/jensimmons/cssremedy). (_Note: If
+  making the switch from Accoutrement/Init to CSS Remedy, you may notice changes
+  in vertical spacing and the displaying of SVGs._)
```
This is very helpful. Is there anything we could link to -- or hints to provide -- for people making that migration?

dvdherron replied:
@jgerigmeyer Made this a bit [more specific](https://github.com/oddbird/accoutrement/pull/97/commits/bbff01555d64914dd49dc684dec1edd618a16567). Does this work better?

jgerigmeyer commented on CHANGELOG.md:
```diff
@@ -2,110 +2,144 @@
+  making the switch from Accoutrement/Init to CSS Remedy, you may notice changes
+  in vertical spacing and the displaying of SVGs._)
+- BREAKING: The old "core" module has been broken into individual modules
+  for [utilities][utils] (extending several Sass modules),
+  [tokens][token] (the map syntax parser),
+  [variables][vars] (converting Sass variables & maps to CSS custom properties),
+  and [ratios][ratio] (for typographic scales and layout aspect ratios).
```
It seems like maybe we should get rid of these absolute links and replace them with relative links, e.g.:

```suggestion
  for [utilities](utils.html) (extending several Sass modules),
  [tokens](tokens.html) (the map syntax parser),
  [variables](vars.html) (converting Sass variables & maps to CSS custom properties),
  and [ratios](ratio.html) (for typographic scales and layout aspect ratios).
```

dvdherron replied:
I started to do this @jgerigmeyer but then I realized that I had forgotten that people could be reading and interacting with this in the on our site  _and_ here in the repo. For the links to work in both places they would all have to be absolute, right?

jgerigmeyer replied:
Ah, good point! I knew there was a reason why we'd used absolute links in the past 😂

jgerigmeyer reviewed (approved):
Two questions, but neither need to block this. 👍
